### PR TITLE
Apply Grammarly suggestions and fix typos

### PR DESCRIPTION
**Pull request by Surendrajat**
Done:
- [x] README
- [x] analysis
- [x] arch

Since I don't know a lot about `rizin` commands, I may have changed to the wrong wording here and there.

### CODE REVIEW

File: src/arch/8051.md [hunk 1]
```diff
@@ -122,9 +122,9 @@ Set a flag for a variable stored at 0x20 in internal RAM:
 Adding support for new 8051 variants
 ------------------------------------
 
-Follow these steps to add support for new 8051 variants to rizin.
+Follow these steps to add support for the new 8051 variants to rizin.
 
-1. Clone latest version of rizin
-2. In '/libr/analysis/p/analysis_8051.c' add a new entry to array 'cpu_models[]' to define a name and a memory mapping. The name of the last entry in array must be NULL
+1. Clone the latest version of rizin
+2. In '/libr/analysis/p/analysis_8051.c' add a new entry to array 'cpu_models[]' to define a name and a memory mapping. The name of the last entry in the array must be NULL
```
Comment (Surendrajat) [inline src/arch/8051.md:128]: Is this `libr` ok here? or should it be `librz`?

File: src/analysis/code_analysis.md [hunk 2]
```diff
@@ -224,19 +224,19 @@ check our work:
...
+There are two very important commands for this: `afc` and `afB`. The latter is a must-know command for some platforms like ARM. It provides a way to change the "bitness" of a particular function. Basically, allowing to select between ARM and Thumb modes.
 
 `afc` on the other side, allows to manually specify function calling convention. You can find more information on its usage in [calling_conventions](calling_conventions.md).
 
 ## Recursive analysis
 
-There are 5 important program wide half-automated analysis commands:
+There are 5 important program-wide half-automated analysis commands:
 
- - `aab` - perform basic-block analysis ("Nucleus" algorithm)
```
Comment (Surendrajat) [inline src/analysis/code_analysis.md:235]: Is this space needed here?

File: src/analysis/emulation.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ For partial emulation (or imprecise full emulation) rizin uses its own
 [ESIL](../disassembling/esil.md) intermediate language and virtual machine.
 
 Rizin supports this kind of partial emulation for all platforms that
-implement ESIL uplifting (x86/x86_64, ARM, arm64, MIPS, powerpc, sparc, AVR, 8051, Gameboy, ...).
+implement ESIL uplifting (x86/x86_64, ARM, arm64, MIPS, PowerPC, Sparc, AVR, 8051, Gameboy, ...).
```
Comment (XVilka) [inline src/analysis/emulation.md:15]: I think it's SPARC, an abbreviation:
> SPARC (Scalable Processor Architecture) 

https://en.wikipedia.org/wiki/SPARC

File: src/arch/8051.md (hunk 1 above)
Comment (XVilka) [reply src/arch/8051.md:128]: Should be indeed `librz/`

File: src/arch/8051.md [hunk 4]
```diff
@@ -122,9 +122,9 @@ Set a flag for a variable stored at 0x20 in internal RAM:
...
 3. In 'libr/asm/p/asm_8051.c' append entry with the same name to '.cpus' attribute
```
Comment (XVilka) [inline src/arch/8051.md:129]: Please do the same here.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by officialcjunior** [comment]
Hello @Surendrajat, I think you can also apply Grammarly on Rizin's README.
I believe that it'll come up with a lot of good suggestions.
Just a thought 🙂

**Comment by Surendrajat** [comment]
@officialcjunior sure thing! Will do that tomorrow.

**Comment by Surendrajat** [comment]
@officialcjunior seems like README is well written. Grammarly did not find much. Anyway will check all other files.

**Review by officialcjunior — APPROVED** [review]

**Comment by Surendrajat** [comment]
@XVilka @ITAYC0HEN let's merge these, I'll finish rest in separate PR.

**Review by ITAYC0HEN — APPROVED** [review]

**Comment by ITAYC0HEN** [comment]
Looks good to me :)
@xvilka merge it if the changes you requested were applied

**Review by XVilka — APPROVED** [review]

**Pull request merged**